app/controllers/posts_controller.rb

```diff
@@ -0,0 +1,55 @@
+class PostsController < ApplicationController
+before_action :find_post, only: [:show, :edit, :update, :destroy]
+
+  def index
+    @posts = Post.all.order(created_at: :desc)
+  end
+
+  def new
+    @post = Post.new
+  end
+
+  def create
+    @post = Post.new post_params
+
+    if @post.save
+      # redirect_to @post
+      redirect_to posts_path
+
+    else
+        render :new
+      end
+  end
+
+  def show
```

Comment:
If these actions aren't used in the routes then you don't need to/shouldn't have them in your controllers

Reply:
Matt,

All actions are in the routes as I could see. Could you see which actions don't have action related in the routes?